File: app/src/main/java/com/ebsoft/watchlist/data/model/db/Watchlist.java
package com.ebsoft.watchlist.data.model.db;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import org.apache.commons.lang3.mutable.MutableInt;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by endre on 09/09/18.
 */

@Entity(tableName = "watchlist")
public class Watchlist implements Serializable {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    @NonNull
    private int id;

    @ColumnInfo(name = "name")
    @NonNull
    private final String name;

    @Ignore
    private final MutableInt mStockCount;

    public Watchlist(String name) {
        this.name = name;
        this.mStockCount = new MutableInt();
    }

    @NonNull
    public String getName() {
        return name;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getStockCount() {
        return mStockCount.intValue();
    }

    public void setStockCount(int count) {
        this.mStockCount.setValue(count);
    }
}

File: app/src/main/java/com/ebsoft/watchlist/data/model/db/Watchlist.kt
package com.ebsoft.watchlist.data.model.db

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore
import android.arch.persistence.room.PrimaryKey
import android.databinding.ObservableInt
import java.io.Serializable

/**
 * Created by endre on 09/09/18.
 */

@Entity(tableName = "watchlist")
data class Watchlist @Ignore constructor (
        @ColumnInfo(name = "name")
        var name: String = "",

        @PrimaryKey(autoGenerate = true)
        @ColumnInfo(name = "id")
        var id: Int = 0,

        @Ignore
        val stockCountObservable: ObservableInt = ObservableInt(0)) : Serializable {

        constructor(name: String) : this(name, 0, ObservableInt(0))
}
